codex/sales/salescontext.nim

```diff
@@ -11,15 +14,18 @@ type
     onStore*: ?OnStore
     onClear*: ?OnClear
     onSale*: ?OnSale
-    onSaleErrored*: ?OnSaleErrored
     proving*: Proving
+    reservations*: Reservations
+  # TODO: do not declare BatchProc as it's declared in node, but causes a
+  # circular dep if node is imported
+  BatchProc* = proc(blocks: seq[bt.Block]): Future[void] {.gcsafe, upraises:[].}
```

Comment:
Perhaps we can move this definition into a separate module, that is imported in both `node.nim` and `salescontext.nim`?

Reply:
Done.